File: Cliente/Windows/HardwareCollector/Conexion/ComandoMaquinaNueva.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web.Script.Serialization;

namespace HardwareCollector.Conexion
{
    public class ComandoMaquinaNueva: Comando
    {
        public Datos datos;
        public ComandoMaquinaNueva() : base("maquina_nueva") {
            datos = new Datos();
        }

        public class Datos
        {
            public string nombre_maquina;
            public SistemaOperativo sistema_operativo;
            public Datos()
            {
                nombre_maquina = "";
                sistema_operativo = new SistemaOperativo();
            }
        }

        public class SistemaOperativo {
            public string nombre;
            public string version;
            public SistemaOperativo() {
                nombre = "";
                version = "";
            }
        }

        public override string Serialize()
        {
            return new JavaScriptSerializer().Serialize(this);
        }

        public override Comando Deserialize(string json)
        {
            //return (ComandoMaquinaNueva)new JavaScriptSerializer().Deserialize(json, this.GetType());
            return (Comando)new JavaScriptSerializer().Deserialize(json, this.GetType());
        }
        
    }
}

File: Cliente/Linux/conexion/comando_maquina_nueva.py
from conexion.comando import Comando
import json

class ComandoMaquinaNueva(Comando):
    
    def __init__(self):
        super().__init__("maquina_nueva")
        self.datos = self.Datos()

    def get_datos(self):
        return self.datos

    class Datos:
        def __init__(self):
            self.nombre_maquina = ""
            self.sistema_operativo = None

        def actualizar_datos(self, maquina):
            self.nombre_maquina = maquina.getnombre()
            self.sistema_operativo = ComandoMaquinaNueva.DatosSistemaOperativo()
            self.sistema_operativo.actualizar_so(maquina.getsistemaoperativo().__dict__)

    class DatosSistemaOperativo:
        def __init__(self):
            self.nombre = ""
            self.version = ""

        def actualizar_so(self, dict_elemento):
            self.__dict__.update(dict_elemento)

    def serialize(self):
        datos_so = self.datos.sistema_operativo.__dict__
        self.datos.sistema_operativo = datos_so
        datos = self.datos.__dict__
        self.datos = datos
        return json.dumps(self.__dict__)
